sdk/python/kfp/compiler/compiler.py

```diff
@@ -297,10 +302,16 @@ def _get_dependencies(self, pipeline, root_group, op_groups):
         dependencies[downstream_groups[0]].add(upstream_groups[0])
     return dependencies
 
-  def _resolve_value_or_reference(self, value_or_reference, inputs):
+  def _resolve_value_or_reference(self, value_or_reference, potential_references):
+    """_resolve_value_or_reference resolves values and PipelineParams, which could be task parameters or input parameters.
+
+    Args:
+      value_or_reference: value or reference to be resolved. It could be basic python types or PipelineParam
+      potential_references(dict{str->str}): a dictionary of parameter names to task names
```

Comment:
Thanks for clarifying this.
I think it can be made even more clear. Why are the references potentials? What kind parameter names are these?

P.S. The python dictionary type is:`from typing import Mapping`, `potential_references: Mapping[str, str]`. Is the type correct though?